tools/functional-tester/etcd-tester/failure_agent.go

```diff
@@ -29,11 +29,31 @@ const (
 	// Otherwise, we get different hash values from the previous revision.
 	// For more detail, please see https://github.com/coreos/etcd/issues/5121.
 	waitRecover = 5 * time.Second
+
+	// waitUntilEtcdIsAlive() checks etcd node aliveness every 100 ms for a maximum of 10 second
+	// immediately after recoverStop() restarts etcd node.
+	// this wait ensures that etcd is alive before recoverStop() returns.
+	// For more detail, please see https://github.com/coreos/etcd/issues/6654.
+	waitEtcdIsAlive = 100 * time.Millisecond
 )
 
 func injectStop(m *member) error { return m.Agent.Stop() }
 func recoverStop(m *member) error {
 	_, err := m.Agent.Restart()
+	if err != nil {
+		return err
+	}
+	return waitUntilEtcdIsALive()
```

Comment:
@fanminshi this wont work. finishing recovering one etcd member does not mean the cluster is recovered. for example, if we kill the all of members, we need to wait at least for the majority to be alive to finish the check. i think we should range over all members after the full recovery instead of doing this one member by another.